bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,625 @@
+                        }
+                    };
+                    const dataServer = cloneJSON(data);
+
+                    this.updateLonLat(cloneJSON(data));
+
+                    if (event.isPaused() && this.preciseTransform) {
+                        this.getTransformedCoordinatesFromServer(dataServer, false, true);
+                    }
+
+                    if (event.isPaused() && this.config.isReverseGeocode) {
+                        this.updateReverseGeocode(cloneJSON(data));
+                    }
+
+                    if (event.isPaused()) {
+                        this.getEmergencyCallInfo(cloneJSON(data));
+                    }
+                }
+            },
+            /**
+                 * @method AfterMapMoveEvent
+                 * Shows map center coordinates after map move
+                 */
+            AfterMapMoveEvent: function (event) {
+                if (this.state.showMouseCoordinates) {
+                    this.updateState({
+                        loading: true
+                    });
+
+                    if (this.preciseTransform) {
+                        this.getTransformedCoordinatesFromServer(null, false, true);
+                    } else {
+                        this.updateLonLat();
+                    }
+                }
+                this.getEmergencyCallInfo();
+            },
+            /**
+             * @method MapClickedEvent
+             * @param {Oskari.mapframework.bundle.mapmodule.event.MapClickedEvent} event
+             */
+            MapClickedEvent: function (event) {
+                const lonlat = event.getLonLat();
+                const data = {
+                    'lonlat': {
+                        'lat': parseFloat(lonlat.lat),
+                        'lon': parseFloat(lonlat.lon)
+                    }
+                };
+                const dataServer = cloneJSON(data);
+                if (!this.showMouseCoordinates) {
+                    if (this.preciseTransform) {
+                        this.getTransformedCoordinatesFromServer(dataServer, false, true);
+                    } else {
+                        this.updateLonLat(data);
+                    }
+                }
+
+                this.getEmergencyCallInfo(cloneJSON(data));
+            }
+        };
+        Object.getOwnPropertyNames(handlers).forEach(p => this.sandbox.registerForEventByName(this, p));
+        return handlers;
+    }
+
+    onEvent (e) {
+        var handler = this.eventHandlers[e.getName()];
+        if (!handler) {
+            return;
+        }
+
+        return handler.apply(this, [e]);
+    }
+}
+
+const wrapped = controllerMixin(UIHandler, [
+    'toggleMouseCoordinates',
+    'showDegrees',
+    'showPopup',
+    'markersSupported',
+    'setMarker',
+    'centerMap',
+    'setLat',
+    'setLon',
+    'getCrsText',
+    'getPreciseTransform',
+    'setLoading',
+    'getSupportedProjections',
```

Comment:
Controller shouldn't have getters. We should pass getSupportedProjections() etc as props to the React-component.